specification/sql/resource-manager/Microsoft.Sql/preview/2022-02-01-preview/ManagedInstanceDtcs.json

```diff
@@ -0,0 +1,382 @@
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "Gets a list of the managed instance DTC settings.": {
+            "$ref": "./examples/ManagedInstanceDtcList.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Sql/managedInstances/{managedInstanceName}/dtc/{dtcName}": {
+      "get": {
+        "tags": [
+          "ManagedInstanceDtcs"
+        ],
+        "description": "Gets the managed instance DTC settings.",
+        "operationId": "ManagedInstanceDtcs_Get",
+        "parameters": [
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ResourceGroupParameter"
+          },
+          {
+            "$ref": "#/parameters/ManagedInstanceNameParameter"
+          },
+          {
+            "name": "dtcName",
+            "in": "path",
+            "description": "The name of the managed instance DTC.",
+            "required": true,
+            "type": "string",
+            "enum": [
+              "current"
+            ],
+            "x-ms-enum": {
+              "name": "DtcName",
+              "modelAsString": true
+            }
+          },
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successfully retrieved the specified managed instance DTC settings.",
+            "schema": {
+              "$ref": "#/definitions/ManagedInstanceDtc"
+            }
+          },
+          "default": {
+            "description": "*** Error Responses: ***\n\n * 400 InvalidDtcName - User tried to interact with Azure SQL Managed Instance DTC settings with a DTC name different from 'current'.\n\n * 400 InvalidUpdateDtcOperationParameters - The provided set of parameters in the body of the request is either empty or invalid.\n\n * 400 InvalidDtcAuthenticationMethod - DTC authentication method \"{0}\" is not supported.\n\n * 400 InvalidDtcVirtualClusterVersion - Virtual cluster version of the Azure SQL Managed Instance \"{0}\" is not supported.\n\n * 400 ManagementServiceManagedInstanceDtcFeatureDisabled - User tried to interact with Azure SQL Managed Instance DTC settings but DTC feature is currently not supported.\n\n * 404 SubscriptionNotFound - The requested subscription was not found.\n\n * 404 ResourceNotFound - The requested resource was not found."
+          }
+        },
+        "x-ms-examples": {
+          "Gets the managed instance DTC settings.": {
+            "$ref": "./examples/ManagedInstanceDtcGet.json"
+          }
+        }
+      },
+      "put": {
```

Comment:
It is a proxy resource. As explained in previous comments, we only support the Update scenario for DTC resource. The user cannot create or delete the resource by themselves. They can only update it. We want to have PUT request in order to support ARM template deployment.

Reply:
There are multiple comments and haven't seen a specific "proxy" reference, but I got it now.

PUT is for Upsert (Update or Insert or no-op if the resource body didn't change) and PATCH to update certain pieces of your resource (e.g. tags on tracked resources) - that's the appropriate terminology, do you actually need PATCH? Can you just use PUT to update your resource? Both of your request bodies (PUT and PATCH) seem identical

Reply:
Patch uses ManagedInstanceDtcResourceProperties and that is not a proxy resource. PUT uses ManagedInstanceDtcResource which is a proxy resource and has a field called "properties" which is of the ManagedInstanceDtcResourceProperties, so PUT uses a "superset" including fields from ProxyResource.
We have added PATCH because one of the previous reviewers commented that this update operation looks more like a PATCH request, so we've added that functionality too. However we need to retain PUT for ARM template deployment.

Reply:
The body of the PUT request is ManagedInstanceDtc object which can be found in the definitions section of the swagger, and the body of the PATCH request is ManagedInstanceDtcProperties which is defined also in the definitions. There is a subtle difference in the examples - body of PATCH is: {dtcEnabled: true}, while body of PUT is {properties: {dtcEnabled: true}}.

Reply:
Your PUT body references this type: [https://github.com/Azure/azure-rest-api-specs/blob/805e16a53f0a725471e0caa6007b48986c7722d9/specification/common-types/resource-management/v1/types.json#L9](https://github.com/Azure/azure-rest-api-specs/blob/805e16a53f0a725471e0caa6007b48986c7722d9/specification/common-types/resource-management/v1/types.json#L9) and all its properties are marked as readOnly (meaning are not meant to be updatable by a client), so technically your two definitions (PUT and PATCH) end up being identical because you are exposing the same set of updatable properties. 

IMO it seems like PATCH is not really needed, unless you want to update only a smaller subset of properties, but since you are technically not creating a resource, it doesn't seem necessary to expose a PATCH, and instead simply expose all your updatable properties via PUT - hope it makes sense

Reply:
I agree, since they would be identical, I've removed the PATCH oepration.